src/main/java/com/amazon/corretto/hotpatch/Logger.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.corretto.hotpatch;
+
+import java.security.AccessControlException;
+
+public class Logger {
+    public static final String VERBOSE_PROPERTY_NAME = "log4jFixerVerbose";
+    private static boolean verbose = true;
+
+    public static void log(final String message) {
+        if (verbose) {
+            System.out.println(message);
+        }
+    }
+
+    public static void log(final Exception ex) {
+        if (verbose) {
+            ex.printStackTrace(System.out);
+        }
+    }
+
+    public static void setVerbose(boolean staticAgent, String args) {
+        verbose = args == null || !args.contains(VERBOSE_PROPERTY_NAME + "=false");
+        // Override verbose setting based on System property. This option is only for static agents, agents that are
+        // attached to the VM will always have a value for the VERBOSE_PROPERTY_NAME in the args that we want to keep.
+        if (staticAgent) {
+            try {
+                if (System.getProperties().contains(VERBOSE_PROPERTY_NAME)) {
+                    verbose = Boolean.parseBoolean(System.getProperty(VERBOSE_PROPERTY_NAME, "true"));
+                }
+            } catch (Exception e) {
+                // nothing to do here, ensure we don't fail due to SecurityManager
+            }
```

Comment:
We gave instructions to team to call with -javaagent:/path/to/jar=log4jFixerVerbose=false, so it does come through the args and not at a property. The  property didn't work as an agent I don't think the property has be processed yet.


```
$ 'java' -javaagent:/path/Log4jHotPatch.jar -Dlog4jFixerVerbose=false -cp log4j-core-2.12.1.jar:log4j-api-2.12.1.jar:. Vuln
Loading Java Agent version 1 (using ASM8).
Transforming org/apache/logging/log4j/core/lookup/JndiLookup
05:08:07.655 [main] ERROR Vuln - Patched JndiLookup::lookup()

$ java' -javaagent:/path/Log4jHotPatch.jar=log4jFixerVerbose=false -cp log4j-core-2.12.1.jar:log4j-api-2.12.1.jar:. Vuln
05:08:58.772 [main] ERROR Vuln - Patched JndiLookup::lookup()
```

Reply:
The property is there, but doing `System.getProperties().contains(str)` is what fails (always return false). Corrected this code and added tests for it, this is the matrix of configurations:

```
******************
Running AgentMode Verbose Tests
------------------
Test verbose config for Prop:unset Arg:unset Expected:true  Output:true
Test verbose config for Prop:unset Arg:true  Expected:true  Output:true
Test verbose config for Prop:unset Arg:false Expected:false Output:false
Test verbose config for Prop:true  Arg:unset Expected:true  Output:true
Test verbose config for Prop:true  Arg:true  Expected:true  Output:true
Test verbose config for Prop:true  Arg:false Expected:false Output:false
Test verbose config for Prop:false Arg:unset Expected:false Output:false
Test verbose config for Prop:false Arg:true  Expected:true  Output:true
Test verbose config for Prop:false Arg:false Expected:false Output:false
Test passed.
```